library/lua/gui/widgets.lua

```diff
@@ -2026,11 +2027,17 @@ function FilteredList:setFilter(filter, pos)
                 -- start matches at non-space or non-punctuation. this allows
                 -- punctuation itself to be matched if that is useful (e.g.
                 -- filenames or parameter names)
-                if key ~= '' and
+                if key ~= '' then
+                    if self.case_sensitive and
                         not search_key:match('%f[^%p\x00]'..key) and
                         not search_key:match('%f[^%s\x00]'..key) then
-                    ok = false
-                    break
+                            ok = false
+                            break
+                    elseif not string.lower(search_key):match('%f[^%p\x00]'..string.lower(key)) and
+                        not string.lower(search_key):match('%f[^%s\x00]'..string.lower(key)) then
+                            ok = false
+                            break
+                    end
```

Comment:
could we factor out the lowercasing and leave the rest of the code un-duplicated? every time a regular expression is duplicated in code, a magical forest creature dies.